backend/lm_backend/api/cruds/configuration.py

```diff
@@ -0,0 +1,56 @@
+"""Feature CRUD class for SQLAlchemy models."""
+from fastapi import HTTPException
+from loguru import logger
+from sqlalchemy import delete
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from lm_backend.api.cruds.generic import GenericCRUD
+from lm_backend.api.models.feature import Feature
+from lm_backend.api.models.license_server import LicenseServer
+from lm_backend.api.schemas.configuration import ConfigurationCompleteUpdateSchema
+
+
+class ConfigurationCRUD(GenericCRUD):
+    """Configuration CRUD module to implement configuration update."""
+
+    async def delete_features(
+        self, db_session: AsyncSession, configuration_id: int, payload: ConfigurationCompleteUpdateSchema
+    ):
+        """
+        Delete all features that aren't in the payload.
+        """
+        features_in_payload = [feature.id for feature in payload.features] if payload.features else []
+        delete_unused_features_query = (
+            delete(Feature)
+            .where(~Feature.id.in_(features_in_payload))
+            .where(Feature.config_id == configuration_id)
+        )
+
+        try:
+            await db_session.execute(delete_unused_features_query)
+        except Exception as e:
+            logger.error(e)
+            raise HTTPException(status_code=400, detail="Feature could not be deleted.")
+
+    async def delete_license_servers(
+        self, db_session: AsyncSession, configuration_id: int, payload: ConfigurationCompleteUpdateSchema
+    ):
+        """
+        Delete all license servers that aren't in the payload.
+        """
+        license_servers_in_payload = (
+            [license_server.id for license_server in payload.license_servers]
+            if payload.license_servers
+            else []
+        )
```

Comment:
Could shorten this a little more even:

```suggestion
        license_servers_in_payload = [ls.id for ls in payload.license_servers or []]
```

Reply:
Mypy didn't like it, the explict `"if payload.license_servers"` ensures that it's an iterable.